lib/liquid_voting/voting.ex

```diff
@@ -24,31 +25,44 @@ defmodule LiquidVoting.Voting do
 
   """
   def create_vote(attrs \\ %{}) do
-    Repo.transaction(fn ->
-      # TODO: refactor case statements into small functions.
+    Tracer.with_span "#{__MODULE__} #{inspect(__ENV__.function)}" do
+      Tracer.set_attributes([
+        {:request_id, Logger.metadata()[:request_id]},
+        {:vars,
+         [
+           {:organization_id, attrs[:organization_id]},
+           {:participant_email, attrs[:participant_email]},
+           {:proposal_url, attrs[:proposal_url]},
+           {:yes, attrs[:yes]}
+         ]}
+      ])
 
-      %Vote{}
-      |> Vote.changeset(attrs)
-      |> Repo.insert()
-      |> case do
-        {:ok, vote} ->
-          if delegation =
-               Repo.get_by(Delegation,
-                 delegator_id: attrs[:participant_id],
-                 organization_id: attrs[:organization_id]
-               ) do
-            case Delegations.delete_delegation(delegation) do
-              {:ok, _delegation} -> vote
-              {:error, changeset} -> Repo.rollback(changeset)
+      Repo.transaction(fn ->
+        # TODO: refactor case statements into small functions.
```

Comment:
I think refactoring is always in our minds, we don't need to add a TODO. These tend to be there forever and just add noise

Reply:
Yes, this is an old TODO. Will remove.